File: verification/verify_stars.js
const { test, expect, chromium } = require('@playwright/test');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  try {
    // Navigate to the app
    await page.goto('http://localhost:3000');

    // Wait for the game canvas to load
    const canvas = page.getByRole('img', { name: /Space shooter game canvas/i });
    await expect(canvas).toBeVisible({ timeout: 10000 });

    // Wait for a few seconds to let the stars render and drift
    await page.waitForTimeout(3000);

    // Take a screenshot
    await page.screenshot({ path: 'verification/stars_optimized.png' });
    console.log('Screenshot taken: verification/stars_optimized.png');

  } catch (error) {
    console.error('Error:', error);
    process.exit(1);
  } finally {
    await browser.close();
  }
})();

File: verification/verify_stars.py
from playwright.sync_api import sync_playwright, expect
import time
import os

def test_stars(page):
    page.goto("http://localhost:3001")

    # Wait for the Mission Brief to appear, indicating app is loaded
    expect(page.get_by_text("Mission Brief")).to_be_visible()

    # Wait for canvas to render
    time.sleep(2)

    # Take screenshot
    os.makedirs("/app/verification", exist_ok=True)
    page.screenshot(path="/app/verification/stars_verification.png")
    print("Screenshot taken at /app/verification/stars_verification.png")

if __name__ == "__main__":
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()
        try:
            test_stars(page)
        finally:
            browser.close()
